### ISSUE TITLE
test

### ISSUE THREAD

**Issue by an1018** [naoto0804/pytorch-inpainting-with-partial-conv#14]
![image](https://user-images.githubusercontent.com/19808900/45019341-a9623480-b05e-11e8-931b-b59f4bdde08c.png)

Why do I use your code to test that the output image has a mask edge and more than one part, such as  red circles？

**Comment by wangcx2018** [comment]
To run train.py,how to set the datasets of Places2? Need to download in advance？if yes, where do the datasets place? thanks!

**Comment by naoto0804** [comment]
@wangcx2018 It's pretty huge dataset developed by the following [paper](http://places2.csail.mit.edu/PAMI_places.pdf). Please follow the instructions on their website.
I expect to have `data_large`, `val_large`, and `test_large` as the sub-directory for one directory.
Specify the path of the Places2 dataset using `--root /path/to/Places2` in `train.py`

**Comment by naoto0804** [comment]
@an1018: This is the unofficial implementation by the third party and I admit that there still remains some bugs. I really appreciate it if you could report them.

**Comment by naoto0804** [comment]
@wangcx2018 see #15

**Comment by an1018** [comment]
@naoto0804 So why do you think this is happening?

**Comment by naoto0804** [comment]
@an1018 The biggest difference is the network itself. It might be better to strictly follow the paper (they use unet_256 used in pix2pix), but I have difficulty in interpreting some parts of the paper (e.g., `The last partial convolution layer’s input will contain the concatenation of the original input image with hole and original mask.`), since unet_256 does not concatenate the original input image .

**Comment by naoto0804** [comment]
Another possibility is that the mask that I used for this experiment is not suitable.
Recently, the authors make some part of the masks [available](http://masc.cs.gmu.edu/wiki/uploads/partialconv/mask.zip) on their [project site](http://masc.cs.gmu.edu/wiki/partialconv).
Splitting that masks into the training and test subset, and then train the model may be better.

**Comment by jeejeelee** [comment]
@an1018  Hi, I also encountered this issue, In my personal opinion ,the red  region is checkboard artifacts（ according to the original paper(page 7-8),Have you solved this issue?

**Comment by sunjunlishi** [comment]
@an1018 how do you use test.py;i don't know  how to use

**Comment by an1018** [comment]
@sunjunlishi Just like this:
 CUDA_VISIBLE_DEVICES=2 python test.py

**Comment by sunjunlishi** [comment]
oh,i have change the img  path directly.but the result  is not good  as author's article
![result](https://user-images.githubusercontent.com/11845664/49135784-bd969480-f322-11e8-8b1a-3dd6c64f5597.jpg)

**Comment by sunjunlishi** [comment]
could author supply a better model.?it is important for me.My project is in need of watermarking and recognition

**Comment by naoto0804** [comment]
First, I'm not the author. 
Second, as you can see in the codes, the current checkpoint that I share is trained on Places2 dataset. It is natural that you cannot get as good results as the paper if you try on faces.

**Comment by sunjunlishi** [comment]
oh.thank you verymuch for your reply
